src/economy_base.h

```diff
@@ -22,15 +22,17 @@ extern CargoPaymentPool _cargo_payment_pool;
  * Helper class to perform the cargo payment.
  */
 struct CargoPayment : CargoPaymentPool::PoolItem<&_cargo_payment_pool> {
+	/* CargoPaymentID index member of CargoPaymentPool is 4 bytes. */
+	StationID current_station; ///< The current station
+	CargoID ct;                ///< The currently handled cargo type
```

Comment:
You now lost some context, namely that these variables are not saved. Maybe just add NOSAVE: before the line-comment, like most other places do.